### ISSUE TITLE
bug(MAT-Menu): Mat-Menu onTriggerContextMenu(event) closemenu() method is taking long to finish and openMenu method is called before closemethod() finishes exexcution.

### ISSUE THREAD

**Issue by poonamsinghlm** [angular/components#23231]
#### Reproduction

onTriggerContextMenu(event){
        
        event.preventDefault();
        this.menuX = event.x - 10;
        this.menuY = event.y - 10;
        this.menu.closeMenu() // this takes long time to finish and before this openmenu is called.
        this.menu.openMenu()

    }
 
#### Expected Behavior

expecting appropriate menu to open with every right click on parent div(its list of divs within tree,so each div has diffrent menu ).
expecting previous menu to close and open new menu but due to delay in closemenu method it closes on second right click.

#### Actual Behavior

if hasdrawback is false,rightclick on parent div opens menu on first click but not on second click due to above issue.
have used time delay as temporary  fix.

#### Environment

- Angular:
- CDK/Material:
- Browser(s):
- Operating System (e.g. Windows, macOS, Ubuntu):

**Comment by crisbeto** [comment]
I suspect that it's because the closing sequence waits out the animation. That being said, what is the use case for what you posted above? It seems like you could set these values before the menu is opened the first time.

**Comment by poonamsinghlm** [comment]
Thanks @crisbeto .Is there any example to do this. I have tried disabling animation but no luck.

**Comment by crisbeto** [comment]
I don't think that disabling the animations would help either, because the close event is still going to be asynchronous. What I was trying to figure out with the first question is what the use case is for your snippet, rather than opening the menu with the correct coordinates from the start.
